xarray/core/dataset.py

```diff
@@ -2416,6 +2416,60 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=False):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Both the data and coordinate variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, set_index only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+
+        ordered_dims = self.dims
+        coord_columns = [c for c in self if c in ordered_dims]
+
+        # convert all coordinates into data variables
+        ds = self.reset_index(coord_columns).reset_coords()
+
```

Comment:
You might need to assign default coordinates here for dimensions without coordinates. This might be easiest to do by pulling out default variables from this dataset, e.g.,
```python
ds = self.assign_coords(**{k: ds[k] for k in ds.dims})
```

Reply:
I can't see the motivation for including the default coordinates in the dask dataframe for dimensions without coordinates.  But I have ensured that dimensions without coordinates do not cause a problem.

Reply:
My reasoning here would be that the default coordinates might not be necessary in array form, but in flattened DataFrame form they're probably necessary to work with the data. For example, consider an image with dimensions `('x', 'y', 'channel')` without any labels along x and y. If you don't add the default dimensions, when you flatten the image you can no longer distinguish between the two original orderings (x, y) and (y, x).